packages/form/src/Field/Field.tsx

```diff
@@ -0,0 +1,197 @@
+    );
+  },
+);
+
+type LabelProps = Omit<React.HTMLProps<HTMLLabelElement>, "size"> &
+  Omit<React.ComponentProps<typeof TelegraphText>, "as"> & {
+    as?: React.ComponentProps<typeof TelegraphText>["as"];
+  };
+type LabelRef = React.ElementRef<typeof TelegraphText>;
+
+const Label = React.forwardRef<LabelRef, LabelProps>(
+  ({ as = "label", size = "2", ...props }, forwardedRef) => {
+    const context = React.useContext(FieldContext);
+    return (
+      <TelegraphText
+        htmlFor={context.id}
+        as={as}
+        size={size}
+        color={COLOR.Label[context.state]}
+        {...props}
+        ref={forwardedRef}
+      />
+    );
+  },
+);
+
+type MessageProps = Omit<React.ComponentProps<typeof TelegraphText>, "as"> & {
+  as?: React.ComponentProps<typeof TelegraphText>["as"];
+};
+type MessageRef = React.ElementRef<typeof TelegraphText>;
+
+const Message = React.forwardRef<MessageRef, MessageProps>(
+  ({ as = "span", size = "2", children, ...props }, forwardedRef) => {
+    const context = React.useContext(FieldContext);
+    return (
+      <TelegraphText
+        as={as}
+        size={size}
+        color={COLOR.Message[context.state]}
+        {...props}
+        ref={forwardedRef}
+      >
+        {context.message ? context.message : children}
+      </TelegraphText>
+    );
+  },
+);
+
+type ControlProps = React.HTMLProps<HTMLInputElement> & {
+  error?: boolean | string;
+};
+type ControlRef = HTMLInputElement;
+
+const Control = React.forwardRef<ControlRef, ControlProps>(
+  ({ ...props }, forwardedRef) => {
+    const context = React.useContext(FieldContext);
+    return (
+      <Slot
+        id={context.id}
+        value={context.value}
+        onChange={context.onChange}
+        error={context.state === "error"}
+        disabled={context.state === "disabled"}
+        {...props}
+        ref={forwardedRef}
+      />
+    );
+  },
+);
+
+type DefaultProps = React.ComponentProps<typeof Root> & {
+  label?: string;
+  message?: string;
+} & {
+  control?: {
+    name: "TextField";
+    props?: React.ComponentProps<typeof TextField>;
+  };
+  size?: React.ComponentProps<typeof TextField>["size"];
+  variant?: React.ComponentProps<typeof TextField>["variant"];
+  leadingIcon?: React.ComponentProps<typeof TextField>["leadingIcon"];
+  TrailingAction?: React.ComponentProps<typeof TextField>["TrailingAction"];
+};
+
+const Default = ({
+  control = {
+    name: "TextField",
+    props: {},
+  },
+  label,
+  message,
+  value,
+  onChange,
+  disabled,
+  error,
+  ...props
+}: DefaultProps) => {
+  return (
+    <Root value={value} onChange={onChange} disabled={disabled} error={error}>
+      {label && <Label>{label}</Label>}
+      {control && (
+        <Control>
+          {control.name === "TextField" && (
+            <TextField {...props} {...control.props} />
+          )}
+        </Control>
+      )}
+      {(message || typeof error === "string") && <Message>{message}</Message>}
+    </Root>
+  );
+};
```

Comment:
The `<Field/>` component. Let's you pass in normal things like `label`, `value`, `onChange`. Also lets you decide what form control to use (just text field for now). The `<Control/>` component wraps this so that we can pass values directly to the chosen form control via a [slot](https://www.radix-ui.com/primitives/docs/utilities/slot) in a modular way.